### ISSUE TITLE
Accented characters are not correctly rendered on input fields

### ISSUE THREAD

**Issue by phillipuniverse** [BroadleafCommerce/CommunityStarter#10]
Screenshot [http://imgur.com/a/EP8cr](http://imgur.com/a/EP8cr)
Tested on Chrome and Firefox
Can be reproduced on live demo (blc 5.x)
Steps to reproduce it: 
1) select Change language to French ( in the header )
2) add a product to the cart
3) display the mini cart

The value displayed is the raw html value instead of "Mettre à jour"

**Comment by phillipuniverse** [comment]
The issue is that the Broadleaf message source that we defined does not have a different default encoding, and thus it uses the default from Spring, which is `ISO-8859-1`. The fix is in Broadleaf we need to set the `defaultEncoding` property to `UTF-8`. This is how you can fix it temporarily from an `@Configuration` class:

```java
@Bean
public static BeanPostProcessor defaultMessageSourceEncodingPostProcessor() {
    return new BeanPostProcessor() {
    
        @Override
        public Object postProcessBeforeInitialization(Object bean, String beanName) throws BeansException {
            return bean;
        }
    
        @Override
        public Object postProcessAfterInitialization(Object bean, String beanName) throws BeansException {
            if (BroadleafMergeResourceBundleMessageSource.class.isAssignableFrom(bean.getClass())) {
                ((BroadleafMergeResourceBundleMessageSource) bean).setDefaultEncoding("UTF-8");
            }
            return bean;
        }
    };
}
```

**Comment by linuxidefix** [comment]
@phillipuniverse Thank you for your detailed answer.
 I tested it and put the bean into CoreConfig. But the condition seems to be always false (BroadleafMergeResourceBundleMessageSource.class.isAssignableFrom(bean.getClass())) 

Maybe I am doing something wrong .
![screenshot-localhost 8443-2017-08-07-18-58-00](https://user-images.githubusercontent.com/836182/29037097-79e42458-7ba2-11e7-927d-ed5b488d2d54.png)

**Comment by linuxidefix** [comment]
Took me time to understand what you meant :-) Yes, if I use utf8 characters in properties file with utf8 as default charset ( so no html code), it works perfectly and solves the problem.
@phillipuniverse So should we use UTF8 characters instead of HTML codes in properties files ?
In this case #https://github.com/BroadleafCommerce/BroadleafCommerce/issues/1639 may be not relevant anymore and properties_xx files should be decoded.

![screenshot-localhost 8443 2017-08-08 00-45-38](https://user-images.githubusercontent.com/836182/29050498-a0f052da-7bdc-11e7-8c38-f3c4f07fedde.png)

**Comment by phillipuniverse** [comment]
> @phillipuniverse So should we use UTF8 characters instead of HTML codes in properties files ?
In this case #BroadleafCommerce/BroadleafCommerce#1639 may be not relevant anymore and properties_xx files should be decoded.

Definitely, I think using the UTF-8 variants are likely better. Then you can also use `th:text` instead of `th:utext` which is required to decode the htmlentities (like `&acute;`).

I already merged in your other commit from #14 which now I think might actually be better to go through and change those to their UTF-8 variant. Should be an easy find/replace on our side in those message bundles.

**Comment by phillipuniverse** [comment]
This is now resolved with BroadleafCommerce/BroadleafCommerce#1658.
